src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -534,18 +549,20 @@ private function getReadAccessInfo($class, $property)
         }
 
         if ($this->cacheItemPool) {
-            $item = $this->cacheItemPool->getItem(self::CACHE_PREFIX_READ.str_replace('\\', '.', $key));
+            $item = $this->cacheItemPool->getItem(self::CACHE_PREFIX_READ.str_replace(array('\\', '@'), '.', $key));
```

Comment:
This change supports anonymous classes, which where not correctly escaped when creating `$key` and made two tests fail